packages/core/src/anchor/memory/in-memory-anchor-service.ts

```diff
@@ -97,31 +102,78 @@ class InMemoryAnchorService extends AnchorService {
     }
 
     for (const compositeKey of Object.keys(validCandidates)) {
-      const candidates: Candidate[] = validCandidates[compositeKey]
-
-      let nonce = 0
-      let selected: Candidate = null
-
-      for (const candidate of candidates) {
-        const record = (await this._ceramic.ipfs.dag.get(candidate.cid)).value
-
-        let currentNonce
-        if (DoctypeUtils.isSignedRecord(record)) {
-          const payload = (await this._ceramic.ipfs.dag.get(record.link)).value
-          currentNonce = payload.header?.nonce || 0
-        } else {
-          currentNonce = record.header?.nonce || 0
+      let candidates: Candidate[] = validCandidates[compositeKey]
+      candidates = await Promise.all(candidates.map(async c => {
+        c.partialLog.push(...await this._loadCommitHistory(c.cid))
```

Comment:
Can we move loading the commit history to right after line 94 when we first build the Candidate object?

Reply:
👍